integration-tests/test_registration.py

```diff
@@ -65,3 +65,25 @@ def test_double_registration(insights_client):
         machine_id_new = f.read()
 
     assert machine_id_new == machine_id_old
+
+
+@pytest.mark.parametrize("legacy_upload_value", [True, False])
+def test_register_group_option(insights_client, legacy_upload_value):
```

Comment:
this test fails on Satellite with legacy upload, and it's a known issue (you linked the issue too); hence, I'd mark this combination as known failure -- something like:
```python
@pytest.mark.parametrize("legacy_upload_value", [
    pytest.param(True, marks=pytest.mark.xfail),  # fails on Satellite
    pytest.param(False),
])
```
this will mark it as xfail also on non-Sat, even if it works there; ideally there should be a better logic to add the xfail mark to the `True` parameter only when needed; I'd say to defer this to a later time, the general xfail is IMHO good enough for now

Reply:
Not only on satellite, but also on production the bug exists. In another word, it is a common bug, not specific to satellite env.

Reply:
oh ok, then simply drop the comment about Satellite from my above code snippet :)